apps/aeoracle/src/aeo_state_tree.erl

```diff
@@ -313,25 +311,36 @@ find_oracle_query_tree_ids(OracleId, Tree) ->
 find_oracle_query_ids(OracleId, Tree) ->
     find_oracle_query_ids(OracleId, Tree, id).
 
-find_open_oracle_queries(OracleId, FromQueryId, Max, #oracle_tree{otree = T}) ->
+find_oracle_queries(OracleId, FromQueryId, QueryType, Max, #oracle_tree{otree = T}) ->
     IteratorKey = case FromQueryId of
                       '$first' -> OracleId;
                       _        -> <<OracleId/binary, FromQueryId/binary>>
                   end,
     Iterator = aeu_mtrees:iterator_from(IteratorKey, T),
-    find_open_oracle_queries(Iterator, Max).
+    find_oracle_queries(Iterator, QueryType, Max, []).
 
-find_open_oracle_queries(_Iterator, 0) -> [];
-find_open_oracle_queries(Iterator, N) ->
+find_oracle_queries(Iterator, QueryType, N, Acc) when N > 0 ->
     case aeu_mtrees:iterator_next(Iterator) of
         {Key, Value, NextIterator} when byte_size(Key) > ?PUB_SIZE ->
             Query = aeo_query:deserialize(Value),
-            case aeo_query:is_closed(Query) of
-                false -> [Query | find_open_oracle_queries(NextIterator, N-1)];
-                true  -> find_open_oracle_queries(NextIterator, N)
+            case {QueryType, aeo_query:is_open(Query)} of
+                {open, true} ->
+                    find_oracle_queries(NextIterator, QueryType, N - 1, [Query | Acc]);
+                {open, false} ->
+                    find_oracle_queries(NextIterator, QueryType, N, Acc);
+                {closed, true} ->
+                    find_oracle_queries(NextIterator, QueryType, N, Acc);
+                {closed, false} ->
+                    find_oracle_queries(NextIterator, QueryType, N - 1, [Query | Acc]);
+                {all, _} ->
+                    find_oracle_queries(NextIterator, QueryType, N - 1, [Query | Acc])
             end;
-        _Other -> [] %% Either end_of_table or next Oracle
-    end.
+        %% Either end_of_table or next Oracle
+        _Other ->
+            Acc
```

Comment:
No lists:reverse here? Please comment.

Reply:
Great catch, thanks! Of course fix is needed there 🥇